blocks/feed/feed.js

```diff
@@ -0,0 +1,51 @@
+import {
+  buildBlock, decorateBlock, loadBlock, readBlockConfig,
+} from '../../scripts/lib-franklin.js';
+import { queryIndex } from '../../scripts/scripts.js';
+
+const resultParsers = {
+  cards: (results) => {
+    const blockContents = [];
+    results.forEach((result) => {
+      const resultTitle = document.createElement('div');
+      resultTitle.textContent = result.title;
+      blockContents.push([resultTitle]);
+    });
+    return blockContents;
+  },
+};
+
+/**
+ * Generic feed block decorator
+ * Example block config:
+ * {
+ *  "block-type": "feed",
+ * "category": "news",
+ * "topic": "press-releases",
+ * "count": 3
+ * }
+ * @param {HTMLElement} block
+ */
+export default async function decorate(block) {
+  const blockCfg = readBlockConfig(block);
+  const blockType = blockCfg['block-type'];
+  const queryObj = await queryIndex();
+
+  const results = queryObj.where((el) => {
+    let match = true;
```

Comment:
it would be good if support for custom logic here based on the feed type

Reply:
I think most of the logic can be achieved via formulating index queries. We can add customization support later as the need arises.